test-material/README.md

```diff
@@ -318,3 +318,57 @@ Prelude> replicate 6 10
 ```
 
 ### List Comprehensions
+A comprehension is a way for you to create a more specific set out of a general
+set. Thus a list comprehension allows you to, in a sense, pick and choose a set
+of elements from a list(s) through the use of predicates, or conditions.
+
+A basic list comprehension for a set (list) that contains the first ten even
+number is
+```haskell
+Prelude> [x*2 | x <- [1..10]]
+[2,4,6,8,10,12,14,16,18,20]
+```
+where x is selected from the list [1..10], or a list that contains all the number
+from 1 to 10. And for every element within that list (which is bound to x), you
+double that element's value.
+
+Using the previous example, you can also add in predicates (conditions) to further
+narrow down your result. For example,
+```haskell
+Prelude> [x*2 | x <- [1..10], x*2 >= 10]
+[10,12,14,16,18,20]
+```
+will do the same thing as the previous comprehension, except that this time you
```

Comment:
I would say that this should also be changed to match correctly the predicates that are going on. Since technically, you are generating another list where you don't necessarily say that it is only to 20, but just greater than or equal to 10. 

```diff
 will do the same thing as the previous comprehension, except that this time you
 - are telling Haskell to only get the even numbers from 10 to 20.

will do the same thing as the previous comprehension, except that this time you
+ are telling Haskell to only get the even numbers greater than or equal to 10.
```